DNAnexus_to_igv/views.py

```diff
@@ -86,142 +122,201 @@ def nexus_search(request):
+                    find_dx_002_bams.py"""
+                ))
+                logging.error(IOe)
+
+                return render(
+                    request, 'DNAnexus_to_igv/nexus_search.html', context_dict
+                )
+
+            # select bams matching sample id, return original entry from
+            # JSON by mtahcing against upper name and search term
+            sample_bams = [
+                v for k, v in json_bams.items() if sample_id.upper() in v[0]['bam_name'].upper()
+            ]
 
             if len(sample_bams) == 0:
                 # no bams found
-                messages.add_message(request,
-                                messages.ERROR,
-                                mark_safe(
-                                """Sample {} not found in DNAnexus, either it\
-                                is not available, the full sample name was not\
-                                correctly given or another error has occured.\n 
-                                Please contact the bioinformatics team if you\
-                                believe the sample should be available.""".\
-                                format(sample_id)),
-                                extra_tags="alert-danger"
-                            )
-
-                return render(request, 'DNAnexus_to_igv/nexus_search.html', 
-                                context_dict)
-
-                    
+                messages.add_message(
+                    request,
+                    messages.ERROR,
+                    mark_safe(
+                        """Sample {} not found in DNAnexus, either it is not\
+                        available, the sample name was not correctly\
+                        given or another error has occured. Please contact\
+                        the bioinformatics team if you believe the sample\
+                        should be available.""".format(sample_id)),
+                    extra_tags="alert-danger"
+                )
+                error_log.error((re.sub(
+                    r'\s+', ' ', """Sample {} not found in JSON. Either sample
+                    name mistyped or an error in finding the BAMs for the
+                    sample, possibly missing index.""".format(sample_id)
+                )))
+
+                return render(request, 'DNAnexus_to_igv/nexus_search.html',
+                              context_dict)
+
+
             if len(sample_bams[0]) == 1 and len(sample_bams) == 1:
                 # one sample found with one bam
-
                 # generate the urls
                 bam_url, idx_url = get_dx_urls(
-                                            sample_bams[0][0]["bam_file_id"],
-                                            sample_bams[0][0]["idx_file_id"]
-                                            )
-                
-                # add variables 
+                    request,
+                    sample_id,
+                    sample_bams[0][0]["bam_file_id"],
```

Comment:
from sample bams you take the only element in the list comprehension and you take the first element of that element which is a dict.
Maybe assign a variable like `bam = sample_bams[0]` or even `sample_bams[0][0]`

Reply:
that sensible, changed